### PR TITLE
Add client side form validation

### PR DESCRIPTION
**Pull request by jezdez**
Refs #211, #212, #223

screen shot 2017-02-14 at 16 18 55

screen shot 2017-02-14 at 16 19 25

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: atmo/jobs/forms.py [hunk 1]
```diff
@@ -180,37 +192,29 @@ class Meta(BaseSparkJobForm.Meta):
...
             '<br />Current notebook: <strong>%s</strong>' % self.instance.notebook_name
         )
+        self.fields['start_date'].help_text += (
+            'Changing this field will reset the job schedule. '
+            'Only future dates are allowed.'
+        )
+        self.fields['identifier'].widget.attrs.update({
+            'data-parsley-remote': (
+                reverse('jobs-identifier-available') +
+                '?identifier={value}&id=%s' % self.instance.pk
```
Comment (robhudson) [inline atmo/jobs/forms.py]: If the field is disabled is this still required?

File: package.json [hunk 2]
```diff
@@ -11,12 +11,14 @@
...
-    "jquery": "^2.2.4",
+    "bootstrap-datetime-picker": "^2.4.3",
+    "jasny-bootstrap": "^3.1.3",
+    "jquery": "^3.1.1",
     "marked": "^0.3.6",
     "moment": "^2.17.1",
-    "moment-timezone": "^0.4.0",
+    "moment-timezone": "^0.5.11",
     "notebookjs": "^0.2.6",
+    "parsleyjs": "^2.6.2",
```
Comment (robhudson) [inline package.json]: parsleyjs looks very nice. 👍

File: package.json [hunk 3]
```diff
@@ -11,12 +11,14 @@
     "ansi_up": "^1.3.0",
     "bootstrap": "^3.3.7",
     "bootstrap-confirmation2": "^2.4.0",
-    "eonasdan-bootstrap-datetimepicker": "^4.17.43",
-    "jquery": "^2.2.4",
+    "bootstrap-datetime-picker": "^2.4.3",
+    "jasny-bootstrap": "^3.1.3",
```
Comment (robhudson) [inline package.json]: What's the reason for this extra lib? What does it provide?

File: atmo/jobs/forms.py (hunk 1 above)
Comment (jezdez) [inline atmo/jobs/forms.py]: Oh, right, I totally forgot about this, you're right!

File: package.json (hunk 3 above)
Comment (jezdez) [inline package.json]: That provides the improved file input: http://www.jasny.net/bootstrap/javascript/#fileinput

File: package.json (hunk 2 above)
Comment (jezdez) [inline package.json]: Yeah, it's pretty clean and works well with bootstrap

**Pull request merged**